Feature search functionality

tomanpepe03 opened the pull request.

tomanpepe03 committed:
Create search_venues.html and add to the urls.py

tomanpepe03 committed:
Add basic search algorithm to check if a title contains the search term

tomanpepe03 committed:
Create tutor.html and integrate it into tutor_list.html and search_subjects.html

tomanpepe03 committed:
Add fuzzy search to handle typos in tutor title search

tomanpepe03 committed:
include category in search filtering

tomanpepe03 committed:
create categories.html for list the categories

tomanpepe03 committed:
create categorie_detail.html and connect it with the categories.html

tomanpepe03 committed:
Add Categories to the navbar.html

deakeszter0308 commented on skillswap/settings.py:
The SECRET_KEY should not be hardcoded in the source code. Instead, it should be securely loaded from environment variables (.env file) using os.getenv('SECRET_KEY')
Please create a .env file to store sensitive information like the SECRET_KEY
Example for .env file:
SECRET_KEY=your-secret-key-here
DEBUG=False

deakeszter0308 reviewed (commented):
Everything looks great overall. Well done! I have one small suggestion regarding the SECRET_KEY.
It's a security best practice not to hardcode sensitive information directly in the settings.py file. Instead, you should store it in a .env file to keep it secure.
And one more thing I noticed is that the fuzzywuzzy package is being used in the project, but it isn't listed in the requirements.txt file,could you please add fuzzywuzzy to the requirements.txt?

deakeszter0308 committed:
Update settings.py

Replaced hardcoded SECRET_KEY with environment variable using os.getenv('SECRET_KEY')

tomanpepe03 committed:
Add fuzzywuzzy to requirements.txt

tomanpepe03 committed:
Add fuzzywuzzy to requirements.txt

tomanpepe03 committed:
Fix syntax error
